app/src/ui/preferences/preferences.tsx

```diff
@@ -110,16 +107,16 @@ export class Preferences extends React.Component<
     const availableEditors = editors.map(e => e.editor)
     const availableShells = shells.map(e => e.shell)
 
-    this.setState({
-      committerName,
-      committerEmail,
-      optOutOfUsageTracking: this.props.optOutOfUsageTracking,
-      confirmRepositoryRemoval: this.props.confirmRepositoryRemoval,
-      confirmDiscardChanges: this.props.confirmDiscardChanges,
+    this.setState((_state, props) => ({
+      committerName: committerName || '',
+      committerEmail: committerEmail || '',
+      optOutOfUsageTracking: props.optOutOfUsageTracking,
```

Comment:
Just a heads up, these three props are now failing the linting because `tslint-microsoft-contrib` is not correctly detecting that `props.[x]` is a valid use of the prop. I tested this against the latest package and it's still happening, so that's either going to need some muting (short-term, unblock PR) or getting this fixed upstream (harder, but better for addressing more of these situations).

Reply:
Yep, it looks like it failing because the rule doesn't understand this usage of setState. I have opened an issue: https://github.com/Microsoft/tslint-microsoft-contrib/issues/796

Thanks for taking a look at this!

Reply:
I was able to work around this in a74208d72 by using the props to set the initial state - which is correctly detected by our linting tools. We already did this with some props, so it was easy to justify doing it for these problem props. 

Let's see if that improves things at all.

Reply:
Great to hear. I'll check things on my end to see if that clears up the alerts.